homeassistant/components/shopping_list.py

```diff
@@ -34,36 +35,47 @@
 SERVICE_COMPLETE_ITEM = 'complete_item'
 
 SERVICE_ITEM_SCHEMA = vol.Schema({
-    vol.Required(ATTR_NAME): vol.Any(None, cv.string)
+    vol.Required(ATTR_NAME): vol.Any(None, cv.string),
+    vol.Optional(ATTR_LIST_ID, '0'): cv.string
 })
 
+WS_TYPE_SHOPPING_LIST_LISTS = 'shopping_list/lists'
 WS_TYPE_SHOPPING_LIST_ITEMS = 'shopping_list/items'
 WS_TYPE_SHOPPING_LIST_ADD_ITEM = 'shopping_list/items/add'
 WS_TYPE_SHOPPING_LIST_UPDATE_ITEM = 'shopping_list/items/update'
 WS_TYPE_SHOPPING_LIST_CLEAR_ITEMS = 'shopping_list/items/clear'
 
+SCHEMA_WEBSOCKET_LISTS = \
+    websocket_api.BASE_COMMAND_MESSAGE_SCHEMA.extend({
+        vol.Required('type'): WS_TYPE_SHOPPING_LIST_LISTS
+    })
+
 SCHEMA_WEBSOCKET_ITEMS = \
     websocket_api.BASE_COMMAND_MESSAGE_SCHEMA.extend({
-        vol.Required('type'): WS_TYPE_SHOPPING_LIST_ITEMS
+        vol.Required('type'): WS_TYPE_SHOPPING_LIST_ITEMS,
+        vol.Required('list_id'): str
     })
 
 SCHEMA_WEBSOCKET_ADD_ITEM = \
     websocket_api.BASE_COMMAND_MESSAGE_SCHEMA.extend({
         vol.Required('type'): WS_TYPE_SHOPPING_LIST_ADD_ITEM,
+        vol.Required('list_id'): str,
         vol.Required('name'): str
     })
 
 SCHEMA_WEBSOCKET_UPDATE_ITEM = \
     websocket_api.BASE_COMMAND_MESSAGE_SCHEMA.extend({
         vol.Required('type'): WS_TYPE_SHOPPING_LIST_UPDATE_ITEM,
+        vol.Required('list_id'): str,
```

Comment:
Since I don't see an accompanying frontend PR, this would break the frontend if we merge this.

Reply:
https://github.com/home-assistant/home-assistant-polymer/pull/2164